### PR TITLE
hal: renesas: rzg: Initial support for Interrupt Control

### PR DESCRIPTION
**Pull request by nhutnguyenkc**
Initial HAL support for Interrupt Control

### CODE REVIEW

**Review by thaoluonguw — APPROVED** [review]
It looks good to me.

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*
+* Copyright (c) 2020 - 2024 Renesas Electronics Corporation and/or its affiliates
+*
+* SPDX-License-Identifier: BSD-3-Clause
+*/
+#ifndef R_INTC_IRQ_CFG_H_
+#define R_INTC_IRQ_CFG_H_
+#ifdef __cplusplus
+            extern "C" {
+            #endif
```
Comment (duynguyenxa) [inline zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h:10]: @nhutnguyenkc , Please recheck the code alignment here

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#define R_INTC_IRQ_CFG_H_
+#ifdef __cplusplus
+            extern "C" {
+            #endif
+
+#define INTC_IRQ_CFG_PARAM_CHECKING_ENABLE (BSP_CFG_PARAM_CHECKING_ENABLE)
+
+#ifdef __cplusplus
+            }
+            #endif
```
Comment (duynguyenxa) [inline zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h:16]: Same as above

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
+* Copyright (c) 2020 - 2024 Renesas Electronics Corporation and/or its affiliates
+*
+* SPDX-License-Identifier: BSD-3-Clause
+*/
+#ifndef R_INTC_NMI_CFG_H_
+#define R_INTC_NMI_CFG_H_
+#ifdef __cplusplus
+            extern "C" {
+            #endif
```
Comment (duynguyenxa) [inline zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h:10]: Same as above

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#define R_INTC_NMI_CFG_H_
+#ifdef __cplusplus
+            extern "C" {
+            #endif
+
+#define INTC_NMI_CFG_PARAM_CHECKING_ENABLE (BSP_CFG_PARAM_CHECKING_ENABLE)
+
+#ifdef __cplusplus
+            }
+            #endif
```
Comment (duynguyenxa) [inline zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h:16]: Same as above

**Review by duynguyenxa — CHANGES_REQUESTED** [review]

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h (hunk 1 above)
Comment (nhutnguyenkc) [reply zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h]: @duynguyenxa , I have fixed the alignment. Please help me check again.

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h (hunk 2 above)
Comment (nhutnguyenkc) [reply zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_irq_cfg.h]: I have fixed the alignment. Please help me check again.

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h (hunk 3 above)
Comment (nhutnguyenkc) [reply zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h]: I have fixed the alignment. Please help me check again.

File: zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h (hunk 4 above)
Comment (nhutnguyenkc) [reply zephyr/rz/rz_cfg/fsp_cfg/rzg/r_intc_nmi_cfg.h]: I have fixed the alignment. Please help me check again.

**Review by duynguyenxa — APPROVED** [review]

**Review by KhiemNguyenT — APPROVED** [review]

**Pull request merged**